CLAUDE.md

```diff
@@ -0,0 +1,242 @@
+pkill -f "mlflow server" || true; pkill -f "yarn start" || true
+
+# Start both MLflow backend and React frontend dev servers
+nohup uv run bash dev/run-dev-server.sh > /tmp/mlflow-dev-server.log 2>&1 &
+
+# Monitor the logs
+tail -f /tmp/mlflow-dev-server.log
+
+# Servers will be available at:
+# - MLflow backend: http://localhost:5000
+# - React frontend: http://localhost:3000
+```
+
+This uses `uv` (fast Python package manager) to automatically manage dependencies and run the development environment.
+
+## Development Commands
+
+### Testing
+
+```bash
+# Run Python tests
+uv run pytest tests/
+
+# Run specific test file
+uv run pytest tests/test_version.py
+
+# Run JavaScript tests
+pushd mlflow/server/js && yarn test; popd
+```
+
+### Code Quality
+
+```bash
+# Python linting and formatting with Ruff
+uv run ruff check . --fix         # Lint with auto-fix
+uv run ruff format .              # Format code
+
+# Check for MLflow spelling typos
+uv run bash dev/mlflow-typo.sh .
+
+# JavaScript linting and formatting
+pushd mlflow/server/js && yarn lint; popd
+pushd mlflow/server/js && yarn prettier:check; popd
+pushd mlflow/server/js && yarn prettier:fix; popd
+
+# Type checking
+pushd mlflow/server/js && yarn type-check; popd
+
+# Run all checks
+pushd mlflow/server/js && yarn check-all; popd
+```
+
+### Special Testing
+
+```bash
+# Run tests with minimal dependencies (skinny client)
+uv run bash dev/run-python-skinny-tests.sh
+
+# Test in Docker container
+uv run bash dev/run-test-container.sh
+```
+
+### Documentation
+
+```bash
+# Build documentation site (needs gateway extras for API doc generation)
+uv run --all-extras bash dev/build-docs.sh --build-api-docs
+
+# Build with R docs included
+uv run --all-extras bash dev/build-docs.sh --build-api-docs --with-r-docs
+
+# Serve documentation locally (after building)
+cd docs && yarn serve --port 8080
+```
+
+## Architecture Overview
+
+### Core Components
+
+1. **MLflow Tracking** (`mlflow/tracking/`): Records and queries experiments, runs, parameters, metrics, and artifacts
+2. **MLflow Models** (`mlflow/models/`): General format for packaging models that can be deployed to various serving environments
+3. **MLflow Projects** (`mlflow/projects/`): Packaging format for reproducible runs with dependency management
+4. **MLflow Server** (`mlflow/server/`): REST API server with FastAPI backend and React frontend
+5. **MLflow Tracing** (`mlflow/tracing/`): LLM observability and debugging tools
+
+### Key Directories
+
+- `mlflow/`: Core Python package
+  - `entities/`: Data model definitions
+  - `store/`: Backend storage implementations (file, database)
+  - `pyfunc/`: Python function flavor for models
+  - `server/`: REST API and web UI
+    - `js/`: React-based frontend application
+  - `protos/`: Protocol buffer definitions
+  - `gateway/`: MLflow AI Gateway for LLMs
+
+- `tests/`: Test suite
+- `examples/`: Usage examples for various ML frameworks
+- `docs/`: Documentation source files
```

Comment:
This will pretty quickly become out of date, and there are lots of other important directories too (e.g. tracking). Can we remove or generalize?